examples/usercall-extension-bind/runner/src/main.rs

```diff
@@ -0,0 +1,356 @@
+    fn from_reader(rdr: &mut impl Read) -> IoResult<Self> {
+        let mut sig : [u8; 12] = [0; 12];
+        let _ = rdr.read_exact(&mut sig[..])?;
+        if &sig[..] != HEADER_SIG {
+            return Err(Error::new(ErrorKind::InvalidData, "Protocol header signature mismatch"))
+        }
+        let ver_cmd = rdr.read_u8()?;
+        let fam = rdr.read_u8()?;
+        let len = rdr.read_u16::<NetworkEndian>()?;
+        let addr_reader = match (ver_cmd, fam) {
+            (V2CMD_LOCAL, FAMILY_UNSPEC) => ProxyAddrReader::UNSPEC(len),
+            (V2CMD_PROXY, FAMILY_TCP4) =>   ProxyAddrReader::V4(len),
+            (V2CMD_PROXY, FAMILY_TCP6) =>   ProxyAddrReader::V6(len),
+            _ => Err(Error::new(ErrorKind::InvalidData, "Unsupported version/command/family"))?
+        };
+    
+        Ok(ProxyHdrV2 {
+               sig,
+               ver_cmd,
+               fam,
+               len,
+               addr_reader,
+          })
+    }
+}
+fn read_proxy_protocol_header(stream :&mut TcpStream) -> IoResult<(
+                                                                  Option<SocketAddr>,
+                                                                  Option<SocketAddr>)> {
+
+    let hdr = ProxyHdrV2::from_reader(stream)?;
+    hdr.addr_reader.read(stream)
+}
+
+
+struct ProxyProtocol {
+    listener      : TcpListener,
+    pub local_address : String,
+}
+
+impl ProxyProtocol {
+    fn new(addr: &str)-> IoResult<Self> {
+        TcpListener::bind(addr).map(|listener| {
+                let local_address = match listener.local_addr() {
+                    Ok(local_address) => local_address.to_string(),
+                    Err(_) => "error".to_string(),
+                };
+                ProxyProtocol {listener, local_address}
+        })
+    }
+}
+
+impl SyncListener for ProxyProtocol {
+    fn accept(&self) -> IoResult<(Box<dyn SyncStream>, Box<dyn ToString>, Box<dyn ToString>)> {
+        let (mut stream, peer_address) = self.listener.accept()?;
+        let local_address_str = stream.local_addr().map(|addr| addr.to_string())?;
+        let peer_address_str = peer_address.to_string();
+        eprintln!("runner:: bind - local_address is {}, peer address is {}", local_address_str, peer_address_str);
+        let (local_address, peer_address) = read_proxy_protocol_header(&mut stream)?;
+
+        let local_address = match local_address {
+            Some(local_address) => local_address.to_string(),
+            None => local_address_str.to_string()
+        };
+        let peer_address = match peer_address {
+            Some(peer_address) => peer_address.to_string(),
+            None => peer_address_str.to_string()
+        };
+        Ok((Box::new(stream), Box::new(local_address), Box::new(peer_address)))
+    }
+}
+
+const HAPROXY_ADDRESS : &str ="localhost:6010";
+
+#[derive(Debug)]
+struct ExternalService;
+// Ignoring local_addr and peer_addr, as they are not relevant in the current context.
```

Comment:
What is this comment referring to?

Reply:
Thanks. Removed the comment in recent push.